app/javascript/src/LoginPage/loginForms/SignupForm.jsx

```diff
@@ -20,101 +22,94 @@ type Props = {
   }
 }
 
-type State = {
-  username: string,
-  emailAddress: string,
-  firstname: string,
-  lastname: string,
-  password: string,
-  passwordConfirmation: string,
-  isFilledUsername: ?boolean,
-  isFilledEmailAddress: ?boolean,
-  isFilledFirstname: boolean,
-  isFilledLastname: boolean,
-  isFilledPassword: ?boolean,
-  isFilledPasswordConfirmation: ?boolean
+type Validation = {
+  [string]: ?boolean
 }
 
-const InputFormGroup = (props) => {
-  const { isRequired, type, value, onChange, isValid } = props
-  const inputProps = {
-    value,
-    onChange,
-    autoFocus: null,
-    inputIsValid: isValid
-  }
-  return (
-    <FormGroup isRequired={isRequired}
-      type={type}
-      labelIsValid={isValid}
-      inputProps={inputProps}
-    />
-  )
+type State = {
+  'user[username]': string,
+  'user[email]': string,
+  'user[first_name]': string,
+  'user[last_name]': string,
+  'user[password]': string,
+  'user[password_confirmation]': string,
+  validation: Validation
 }
 
+const formFieldsList = [
+  {isRequired: true, type: 'user[username]'},
```

Comment:
I'm sure this could be avoided with the validate instance crafted and injected in the component